python/ray/data/_internal/stats.py

```diff
@@ -196,6 +258,44 @@ def _get_or_create_stats_actor():
     ).remote()
 
 
+_stats_actor: Optional[_StatsActor] = None
+_stats_actor_cluster_id: Optional[str] = None
+"""This global _stats_actor may be from a previous cluster that was shutdown.
+We store _cluster_id to check that the stored actor exists in the current cluster.
+"""
+
+
+def _check_cluster_stats_actor():
+    # Checks if global _stats_actor belongs to current cluster,
+    # if not, creates a new one on the current cluster.
+    global _stats_actor, _stats_actor_cluster_id
+    current_cluster_id = ray._private.worker._global_node.cluster_id
+    if _stats_actor is None or _stats_actor_cluster_id != current_cluster_id:
+        _stats_actor = _get_or_create_stats_actor()
+        _stats_actor_cluster_id = current_cluster_id
+
+
+def update_stats_actor_metrics(
+    op_metrics: List[OpRuntimeMetrics], tags: Dict[str, str]
+):
+    global _stats_actor
+    _check_cluster_stats_actor()
+
+    metric_keys = OpRuntimeMetrics.get_metric_keys()
+    stats = {key: 0 for key in metric_keys}
+    for op_metric in op_metrics:
+        metric_dict = op_metric.as_dict()
+        stats = {key: stats[key] + metric_dict.get(key, 0) for key in metric_keys}
```

Comment:
nit, update the stats in-place would be more readable and (slightly) more efficient. 
```
for key in metric_keys:
   stats[key] += metric_dict.get(key, 0)
```